src/orbitdb-access-controller.js

```diff
@@ -0,0 +1,134 @@
+'use strict'
+
+const pMapSeries = require('p-map-series')
+const AccessController = require('./access-controller-interface')
+const ensureAddress = require('./utils/ensure-ac-address')
+
+const type = 'orbitdb'
+
+class OrbitDBAccessController extends AccessController {
+  constructor (orbitdb, options) {
+    super()
+    this._orbitdb = orbitdb
+    this._db = null
+    this._options = options || {}
+  }
+
+  // Returns the type of the access controller
+  static get type () { return type }
+
+  // Returns the address of the OrbitDB used as the AC
+  get address () {
+    return this._db.address
+  }
+
+  // Return true if entry is allowed to be added to the database
+  async canAppend (entry, identityProvider) {
+    // Write keys and admins keys are allowed
+    const access = new Set([...this.get('write'), ...this.get('admin')])
+    // If the ACL contains the writer's public key or it contains '*'
+    if (access.has(entry.identity.publicKey) || access.has('*')) {
+      const verifiedIdentity = await identityProvider.verifyIdentity(entry.identity)
+      // Allow access if identity verifies
+      return verifiedIdentity
+    }
+
+    return false
+  }
+
+  get capabilities () {
+    if (this._db) {
+      let capabilities = this._db.all()
+
+      const toSet = (e) => {
+        const key = e[0]
+        capabilities[key] = new Set([...(capabilities[key] || []), ...e[1]])
+      }
+
+      // Merge with the access controller of the database
+      // and make sure all values are Sets
+      Object.entries({
+        ...capabilities,
+        // Add the root access controller's 'write' access list 
+        // as admins on this controller
+        ...{ admin: new Set([...(capabilities.admin || []), ...this._db.access.write])
+        }
+      }).forEach(toSet)
+
+      return capabilities
+    }
+    return {}
+  }
+
+  get (capability) {
+    return this.capabilities[capability] || new Set([])
+  }
+
+  async close () {
```

Comment:
This is not defined in the AC interface, so it doesn't get called anywhere either. Should possibly add this to the AC interface to have "shutdown" function to wind down state or services the AC uses.

Reply:
Agreed, useful for the odb or contract ac

Reply:
Added to the the interface